packages/catalog-model/src/types.ts

```diff
@@ -34,3 +34,31 @@ export type EntityPolicy = {
 };
 
 export type JSONSchema = JSONSchema7 & { [key in string]?: JsonValue };
+
+/**
+ * A complete entity name, with the full kind-namespace-name triplet.
+ */
+export type EntityName = {
+  kind: string;
+  namespace: string;
+  name: string;
+};
+
+/**
+ * A reference by name to an entity, where the kind and/or the namespace can be
+ * left out.
+ *
+ * Left-out parts of the reference need to be handled by the application,
+ * either by rejecting the reference or by falling back to default values.
+ */
+export type CompoundEntityRef = {
+  kind?: string;
+  namespace?: string;
+  name: string;
+};
+
+/**
+ * A reference by name to an entity, either as a compact string representation,
+ * or as a compound reference structure.
+ */
+export type EntityRef = string | CompoundEntityRef;
```

Comment:
I kinda prefer inlining the object form here, so that it's more directly visible in editors. Also reduces the API surface by not introducing a new name

Reply:
it's used in a couple places above if you check it out

Reply:
Yep, was suggesting to not do that too x)